### PR TITLE
Fix entry point discovery on Windows.

### PR DESCRIPTION
**Pull request by phst**
The zipfile module always uses forward slashes, so we shouldn't use
os.path.join here.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Does not include precompiled binaries, eg. `.par` files. See [CONTRIBUTING.md](https://github.com/bazelbuild/rules_python/blob/master/CONTRIBUTING.md) for info
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Entry points aren't found on Windows

Issue Number: N/A

## What is the new behavior?

Entry points should be found on Windows (haven't fully tested it though)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: python/pip_install/extract_wheels/lib/wheel.py [hunk 1]
```diff
@@ -55,7 +55,10 @@ def entry_points(self) -> Dict[str, str]:
             # Calculate the location of the entry_points.txt file
             metadata = self.metadata
             name = "{}-{}".format(metadata.name.replace("-", "_"), metadata.version)
-            entry_points_path = os.path.join("{}.dist-info".format(name), "entry_points.txt")
+            # Note that the zipfile module always uses the forward slash as
+            # directory separator, even on Windows, so don't use os.path.join
+            # here.
```
Comment (thundergolfer) [inline python/pip_install/extract_wheels/lib/wheel.py:60]:
```suggestion
            # here.
            # Ref, for Python 3.10: https://github.com/python/cpython/blob/3.10/Lib/zipfile.py#L355
```

Substantiates the "always" claim.

**Review by thundergolfer — COMMENTED** [review]
Can we add a TODO to use `zipfile.Path` once 3.8 is our minimum supported version? (`zipfile.Path` is new in 3.8)

It would provide nicer code. Something like: 

```python
entry_points_path = zipfile.Path(whl, f"{name}.dist-info/entry_points.txt")

if not entry_points_path.exists():
    return dict()
...
```

**Comment by phst** [comment]
> Can we add a TODO to use `zipfile.Path` once 3.8 is our minimum supported version? (`zipfile.Path` is new in 3.8)
> 

Done

**Review by alexeagle — APPROVED** [review]
thanks!

**Pull request merged**